src/configure/utilities/InputControlProvider.ts

```diff
@@ -0,0 +1,180 @@
+import { MustacheHelper } from "../helper/mustacheHelper";
+import { ExtendedInputDescriptor, ExtendedPipelineTemplate, InputDataType, InputMode } from "../model/Contracts";
+import { AzureSession, ControlType, IPredicate, StringMap } from '../model/models';
+import { DataSourceExpression } from "./DataSourceExpression";
+import { InputUxDescriptor } from "./InputUxDescriptor";
+import { InputUxDescriptorUtility } from "./InputUxDescriptorUtility";
+import { VisibilityHelper } from "./VisibilityHelper";
+
+export class InputControlProvider {
+    private _pipelineTemplate: ExtendedPipelineTemplate;
+    private _inputUxDescriptors: Map<string,InputUxDescriptor>;
+    private _uxInputs: Map<string,ExtendedInputDescriptor>;
+
+    constructor(pipelineTemplateParameters: ExtendedPipelineTemplate, inputs: { [key: string]: any}){
```

Comment:
why name pipeline template argument as `pipelineTemplatePamrameters` ?

Reply:
changed it to pipelineTemplate